### PR TITLE
EXL-300 fixed special views after import

### PR DESCRIPTION
**Pull request by zsso**
- Fixed custom field names
- Added update() method to PluginIserviceItem trait to handle model in custom field update in one methog
- Renamed fields (data_fact, data_exp_fact, total2_black_fact, total2_color_fact) in glpi_plugin_iservice_movements table and updated code
- Replaced new Cartridge() with new PluginIserviceCartridge()
- replaced new Contract() with new PluginIserviceContract()
- Fixed FacturiClient and LastNTickets display
- Fixed GlobalReadCounter view, added index to hmarfa_facrind, imported missing rights
- Removed container id from duplication check on custom field creation

### CODE REVIEW

File: plugin/inc/specialviews/lastntickets.class.php [hunk 1]
```diff
@@ -4,6 +4,7 @@
 namespace GlpiPlugin\Iservice\Specialviews;
 
 use GlpiPlugin\Iservice\Views\View;
+use \CommonITILActor;
```
Comment (hupupu) [inline plugin/inc/specialviews/lastntickets.class.php:7]: Is this a left-over code?

File: plugin/inc/specialviews/emaintenance.class.php [hunk 2]
```diff
@@ -5,6 +5,7 @@
 
 use GlpiPlugin\Iservice\Utils\ToolBox as IserviceToolBox;
 use GlpiPlugin\Iservice\Views\View;
+use \PluginIserviceTicket;
```
Comment (hupupu) [inline plugin/inc/specialviews/emaintenance.class.php:8]: Is this a left-over code?

File: plugin/inc/views/view.class.php [hunk 3]
```diff
@@ -1409,6 +1412,7 @@ protected function generateWidgetBadge($badge_type, $badge_id, $title, $click, $
 
     protected function evalIfFunction($string, $global_variables = []): string
     {
+        $row = $row ?? [];
         if (strpos($string, 'function:') === 0) {
             $eval_before = '';
             foreach (array_keys($global_variables) as $variable_name) {
```
Comment (hupupu) [inline plugin/inc/views/view.class.php:1418]: I don't think so

**Review by hupupu — CHANGES_REQUESTED** [review]

File: plugin/inc/specialviews/emaintenance.class.php (hunk 2 above)
Comment (zsso) [reply plugin/inc/specialviews/emaintenance.class.php:8]: It is used in line 88 and 92.

File: plugin/inc/specialviews/lastntickets.class.php (hunk 1 above)
Comment (zsso) [reply plugin/inc/specialviews/lastntickets.class.php:7]: It is used in line 88 and 92

File: plugin/inc/views/view.class.php (hunk 3 above)
Comment (zsso) [reply plugin/inc/views/view.class.php:1418]: Fixed

File: plugin/inc/specialviews/facturiclient.class.php [hunk 4]
```diff
@@ -215,7 +220,7 @@ protected function getSettings(): array
                         ",
             'default_limit' => 10,
             'show_limit' => Session::haveRight('plugin_iservice_view_facturi_client', UPDATE),
-            'row_class' => 'function:FacturiClient::getRowBackgroundClass($row);',
+            'row_class' => 'function:\GlpiPlugin\Iservice\Specialviews\FacturiClient::getRowBackgroundClass($row_data);',
```
Comment (hupupu) [inline plugin/inc/specialviews/facturiclient.class.php:223]: ?

**Review by hupupu — APPROVED** [review]

File: plugin/inc/specialviews/facturiclient.class.php (hunk 4 above)
Comment (hupupu) [reply plugin/inc/specialviews/facturiclient.class.php:223]: accepted

**Pull request merged**